src/main/webapp/creatives/maps/SimidMap.js

```diff
@@ -0,0 +1,241 @@
+const DEFAULT_MAP_LAT = 37.422004;
+const DEFAULT_MAP_LNG = -122.081402;
+const DEFAULT_ZOOM = 13;
+const DEFAULT_LOCATION_NUM_DISPLAYED = 4;
+const MARKER_SIZE = 25;
+
+export default class SimidMap {
+    constructor() {
+        /**
+         * The LatLng coordinates representing the closest advertised location.
+         * @private {?google.maps.LatLng}
+         */
+        this.activeLocation_ = null;
+        /**
+         * The LatLng coordinates representing the user's current location.
+         * @private {?google.maps.LatLng}
+         */
+        this.currentLocation_ = null;
+        /**
+         * The string representing the search query.
+         * @private {?string}
+         */
+        this.searchQuery_ = null;
+        /**
+         * The desired marker image's string URL.
+         * @private {?string}
+         */
+        this.markerImage_ = null;
+        /**
+         * The DirectionsRenderer object that displays directions from
+         * the given request.
+         * @private @const {!google.maps.DirectionsRenderer}
+         */
+        this.directionsRenderer_ = new google.maps.DirectionsRenderer();
+        /** 
+         * A map object from the Google Maps API.
+         * @private {?google.maps.Map}
+         */    
+        this.map_ = null;
+        /** 
+         * The element from the document where the travel method selector lives
+         * @private {?Object}
+         */  
+        this.travelMethodElement_ = null;
+        /** 
+         * The element from the document where the time display div lives
+         * @private {?Object}
+         */   
+        this.timeDisplayElement_ = null;
+    }
+
+    setTravelMethodElement(element) {
+        this.travelMethodElement_ = element;
+    }
+
+    setTimeDisplayElement(element) {
+        this.timeDisplayElement_ = element;
+    }
+
+    setSearchQuery(query) {
+        this.searchQuery_ = query;
+    }
+
+    setMarkerImage(markerImage) {
+        this.markerImage_ = markerImage
+    }
+
+    setCurrentLocation(coordinates = new google.maps.LatLng(DEFAULT_MAP_LAT, DEFAULT_MAP_LNG)) {
+        this.currentLocation_ = coordinates;
+    }
+
+    getMap() {
+        return this.map_;
+    }
+
+    /**
+     * Loads a map object that currently defaults to a hardcoded location.
+     * @param {!Object} element The div within the main document where the map is to be displayed.
```

Comment:
I believe your can use Element or HTMLElement here. In general the more specific you can be about the object type the better.

Reply:
Fixed